### PR TITLE
Use summary when appropriate

### PR DESCRIPTION
**Pull request by dshanske**
If the response type is expressly set to comment, then this will always return the full content. Otherwise, it will return the content if it is short enough, if not the summary, if not the name, if not nothing.

I debated on the nothing vs a generated excerpt. But if we were to do a generated excerpt, I'd want to discuss how we'd excerpt as a topic. For example, number of characters = MAX_INLINE_MENTION_LENGTH? Try to mirror Pingbacks and get the context around the link to the target? Etc.

### CODE REVIEW

File: includes/Entity/class-item.php [hunk 1]
```diff
@@ -282,15 +282,29 @@ public function get_author( $property = null ) {
...
+
+		// If there is a summary that isn't empty, return that, if not the name.
+		if ( ! empty( $this->summary ) ) {
 			return $this->summary;
-		} elseif ( $this->name ) {
-			return $this->name;
-		} else {
-			return '';
+		} elseif ( ! empty( $this->name ) ) {
+			return $this - name;
```
Comment (pfefferle) [inline includes/Entity/class-item.php:304]: @dshanske this should be `$this->name`?!!

**Review by pfefferle — CHANGES_REQUESTED** [review]

File: includes/Entity/class-item.php (hunk 1 above)
Comment (dshanske) [reply includes/Entity/class-item.php:304]: Yes, phpcs...

File: includes/Entity/class-item.php [hunk 2]
```diff
@@ -282,15 +282,29 @@ public function get_author( $property = null ) {
 	 * @return string
 	 */
 	public function get_content() {
-		if ( $this->content ) {
+		$text_len = $this->str_length( $this->content );
+
+		if ( ( 0 === $text_len ) ) {
+			if ( ! empty( $this->summary ) ) {
```
Comment (pfefferle) [inline includes/Entity/class-item.php:288]: It think `if ( $this->summary ) {` should do the exact same and maybe a bit easier to read?!?

Comment (dshanske) [reply includes/Entity/class-item.php:288]: Okay. I'll put it back to just $this->summary, but what if it isn't a string? Probably should check that too.

Comment (pfefferle) [reply includes/Entity/class-item.php:288]: is that really possible?

**Comment by dshanske** [comment]
Do you think this will address the raised behavior issue?

**Review by pfefferle — APPROVED** [review]

**Pull request merged**